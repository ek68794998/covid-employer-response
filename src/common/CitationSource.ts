export class CitationSource {
	public date?: string;

	public link: string = "";

	public source: string = "";

	public title: string = "";
}
