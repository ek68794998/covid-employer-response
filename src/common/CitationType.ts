export const CitationTypeValues: string[] = [
	"hearsay",
	"statement",
	"publication",
];

export type CitationType = typeof CitationTypeValues[number];

export const getCitationTypeValue = (type?: CitationType): number =>
	type ? CitationTypeValues.indexOf(type) : -1;
