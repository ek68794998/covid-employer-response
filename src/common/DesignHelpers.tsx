import React from "react";

export class DesignHelpers {
	public static materialIcon(iconName: string, className?: string): JSX.Element {
		return <i className={`material-icons ${className || ""}`}>{iconName}</i>;
	}
}
