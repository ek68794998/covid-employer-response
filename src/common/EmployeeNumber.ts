type EmployeeMagnitude = 6 | 9 | null;

export class EmployeeNumber {
	public exactValue: number;

	public magnitude: EmployeeMagnitude;

	public value: number;

	public constructor(value: number) {
		this.exactValue = value;

		this.magnitude = EmployeeNumber.getEmployeeMagnitude(value);

		this.value =
			this.magnitude
				? Math.round(1234567 / Math.pow(10, this.magnitude - 1)) / 10
				: value;
	}

	public get magnitudeLabel(): string | null {
		return this.magnitude === 6 ? "million" : this.magnitude === 9 ? "billion" : null;
	}

	public toString(): string {
		return this.magnitude ? `${this.value} ${this.getEmployeeMagnitudeName()}` : this.value.toLocaleString();
	}

	private static getEmployeeMagnitude(value: number): EmployeeMagnitude {
		if (!value) {
			return null;
		}

		if (value >= 1000000000) {
			return 6;
		}

		if (value >= 1000000) {
			return 9;
		}

		return null;
	}

	private getEmployeeMagnitudeName(): string | null {
		switch (this.magnitude) {
			case 6: return "million";
			case 9: return "billion";
			default: return null;
		}
	}
}