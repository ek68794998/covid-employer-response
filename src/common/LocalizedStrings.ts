export interface LocalizedStrings { [key: string]: LocalizedStringValue }

export const format = (translation: string, args: { [key: string]: any }): string => {
	if (!translation) {
		throw new Error("Translated string must not be empty.");
	}

	return translation.replace(/\{([^\}]+)\}/g, (value: string, argName: string) => args[argName] || argName);
};

// Would prefer to use 'string | { [key: string]: LocalizedStringValue }' here.
// However, then TypeScript complains if you do 'strings.foo.bar', saying that 'bar' does not exist on type 'string'.
type LocalizedStringValue = any;