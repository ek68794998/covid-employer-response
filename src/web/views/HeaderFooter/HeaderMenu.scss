@import "../Colors.scss";

header {
	border-bottom: 1px solid $outline-color;
	font: 400 24px 'Baloo Chettan 2';
	padding: 16px 60px;
	position: fixed;
	width: 100%;

	.header-container {
		display: flex;
		margin: 0 auto;
		max-width: 1440px;
	}

	div#brand {
		flex: 1;
		font-weight: 600;
	}

	a {
		color: inherit;
		opacity: 0.7;
		padding: 4px 8px 6px;
		text-decoration: none;

		&:hover {
			opacity: 1.0;
		}
	}

	ul#menu {
		flex: 0;
		list-style: none;
		margin: 0;
		padding: 0;
		white-space: nowrap;

		li {
			display: inline-block;
			margin-left: 16px;
		}
	}
}