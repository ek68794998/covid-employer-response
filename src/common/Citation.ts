import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
