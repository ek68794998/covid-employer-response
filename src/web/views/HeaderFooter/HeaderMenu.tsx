import React from "react";
import { useSelector } from "react-redux";
import { Link, withRouter } from "react-router-dom";

import { AppState } from "../../state/AppState";
import { getStrings } from "../../state/ducks/localization/selectors";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import "./HeaderMenu.scss";

const HeaderMenu: React.FC = (): React.ReactElement => {
	const strings: LocalizedStrings = useSelector((state: AppState) => getStrings(state));

	return (
		<header>
			<div className="header-container">
				<div id="brand">
					{strings.appTitle}
				</div>
				<ul id="menu">
					<li><Link to="/">{strings.home}</Link></li>
					<li><Link to="/about">{strings.about}</Link></li>
				</ul>
			</div>
		</header>
	);
};

export default withRouter(HeaderMenu) as any;
