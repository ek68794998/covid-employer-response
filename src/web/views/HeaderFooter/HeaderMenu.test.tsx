import React from "react";
import { render } from "react-dom";
import { BrowserRouter } from "react-router-dom";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import HeaderMenu from "./HeaderMenu";

jest.mock("react-redux", () => ({
	useSelector: (): {} => ({
		getStrings: (): LocalizedStrings => ({
			about: "About",
			appTitle: "App Title",
			appTitleShort: "AT",
			home: "Home",
		}),
	}),
}));

describe("<HeaderMenu />", () => {
	test("renders without exploding", () => {
		const div: HTMLDivElement = document.createElement("div");

		render(
			<BrowserRouter>
				<HeaderMenu.WrappedComponent />
			</BrowserRouter>,
			div,
		);
	});
});
