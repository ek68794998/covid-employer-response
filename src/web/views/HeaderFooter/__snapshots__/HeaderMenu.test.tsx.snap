// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderMenu /> renders without exploding 1`] = `
<header>
  <div
    className="HeaderFooter__Container"
  >
    <button
      className="HeaderFooter__OpenHamburgerMenu"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        menu
      </i>
    </button>
    <div
      className="HeaderMenu__HiddenBrand"
      id="brand"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
        title="λô₥è"
      >
        áƥƥ
        <span
          style={
            Object {
              "color": "red",
            }
          }
        >
            [DEV]
        </span>
      </a>
      <div
        className="HeaderMenu__Subtitle"
      >
        áƥƥTïƭℓè
      </div>
    </div>
    <a
      aria-current={null}
      className="HeaderMenu__Link"
      href="/employers"
      onClick={[Function]}
    />
    <a
      aria-current={null}
      className="HeaderMenu__Link"
      href="/about"
      onClick={[Function]}
    >
      áβôúƭ
    </a>
    <a
      className="HeaderMenu__Link"
      href="https://github.com/ekumlin/covid-employer-response/issues/new/choose"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    />
  </div>
</header>
`;
